presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -457,6 +516,79 @@ private ShardInfo createShardInfo(UUID shardUuid, OptionalInt bucketNumber, Path
         return shardDelta(shardUuid, Optional.of(shard));
     }
 
+    @VisibleForTesting
+    Collection<Slice> writeDeltaDeleteFile(long transactionId, OptionalInt bucketNumber, UUID oldShardUuid, Optional<UUID> oldDeltaDeleteShardUuid, BitSet rowsToDelete)
+    {
+        if (rowsToDelete.isEmpty()) {
+            return ImmutableList.of();
+        }
+        // blockToDelete is LongArrayBlock
+        List<Long> columnIds = new ArrayList<>();
+        List<Type> columnTypes = new ArrayList<>();
+        columnIds.add(0L);
+        columnTypes.add(BIGINT);
+
+        // todo one block can't hold one orcfile
+        // At this point, rowsToDelete couldn't be empty
+        boolean oldShardContainToDelete = false;
```

Comment:
Also I think we might need a better variable name as I was confused when I first read `oldShardContainToDelete`. Maybe `needCreateNewDelta`? But before doing that, please look at my comment in `mergeRowsToDelete()`. I don't think tracking this flag is correct. We should always generate new delta or remove the whole file or throw an assert failure.